apps/remix-ide/src/app/panels/file-panel.js

```diff
@@ -165,13 +166,17 @@ module.exports = class Filepanel extends ViewPlugin {
       }
       return
     }
+    const workspaceProvider = this._deps.fileProviders.workspace
+    workspaceProvider.setWorkspace('default_workspace')
     // insert example contracts if there are no files to show
-    this._deps.fileProviders.browser.resolveDirectory('/', async (error, filesList) => {
-      if (error) console.error(error)
-      if (Object.keys(filesList).length === 0) {
-        await this.createWorkspace('default_workspace')
-      }
-      this.getWorkspaces()
+    return new Promise((resolve, reject) => {
+      this._deps.fileProviders.browser.resolveDirectory('/', async (error, filesList) => {
+        if (error) console.error(error)
```

Comment:
you need to `reject` here

Reply:
and `return` too

Reply:
> you need to `reject` here



> and `return` too

it's already returning a new promise

Reply:
I mean instead of doing `if (error) console.error(error)`
maybe this should do if (error) return reject(error)`